src/app/components/itinerary-list/itinerary-list.component.html

```diff
@@ -0,0 +1,50 @@
+<div class="h-24 bg-gradient-to-r from-blue-400 via-purple-300 to-pink-500 bg-white py-24 sm:py-32">
+  <div class="mx-auto max-w-7xl px-6 lg:px-8">
```

Comment:
Since your `tripDetails` object has no guarantee of being populated, you need to have a safety check here.
`<div class="mx-auto max-w-7xl px-6 lg:px-8" *ngIf="tripDetails?.tripId && tripDetails?.tripName && tripDetails?.description">`